### DISCUSSION TITLE
Jexl for additional options in get

### DISCUSSION THREAD

**Discussion by LuisSoares** [GMOD/jbrowse-components#3978] [Q&A] [answered]
Is there a jexl command to get the values of elements that are not part of the feature. Similar to get(feature,'start') is there a command to get other aspects of what is being displayed such as the range of the region that is being plotted?

**Comment by cmdcolin** [comment]
I don't think there is currently a way to access the 'currently displayed region' inside jexl callbacks in the general case. if you can describe your use case i might be able to advise specifically.

in general accessing the 'regions being plotted' is something accessible from the 'client side code' for example with linearGenomeViewModel.visibleLocStrings (https://jbrowse.org/jb2/docs/models/lineargenomeview/#getter-visiblelocstrings)

however, that code is not available on jexl callbacks because renderers are executed in a different context (the RPC context, see figure describing the RPC boundary here https://jbrowse.org/jb2/docs/developer_guides/creating_renderer/)

it requires manual code to 'marshall' data across the RPC boundary. it can be done though, and if you describe your use case, i might be able to help further.

**Comment by LuisSoares** [comment]
Hi Colin, 
Thank you so much for your reply. I was trying to display RNA structure data in the form of Arc plots and while tinkering with the feature heights I manage to get the arcs to look like half circles (the usual way to visualize them, bottom panel), however once I zoom in or zoom out they become distorted (top panel). Perhaps is a math question that can be resolved by a different formula on the arc height parameter (maybe even a pre-calculated value in one of the fields). Again thank you so much for your help and the work you do in Jbrowse.
![image](https://github.com/GMOD/jbrowse-components/assets/7463255/ac0caecd-b5a9-4767-a852-de37d0d9279d)

**Reply by cmdcolin** [reply, marked as answer]
Hi @LuisSoares 
I am not sure if it's what you are using but I created a modification to the LinearArcRenderer code that explicitly can render semi-circles, as the current one indeed tries to fit the arc height to some particular value that is within the display height. With the semi circles, it will just always follow that circular path no matter the 'height' of the arc

code here https://github.com/GMOD/jbrowse-components/pull/3982


its still a pr proposal but  you can try out this branch with `jbrowse create --branch arc_renderer_semicircles`

**Reply by LuisSoares** [reply]
Hi @cmdcolin 

Thank you so much for your help, the branch works flawlessly for displaying the semi circles!.

Luis

**Reply by cmdcolin** [reply]
v2.7.1 now released with this feature added, so you can use the main release zip:)
